Runtime/UI/ScreenManager.cs

```diff
@@ -7,72 +7,69 @@
 
 namespace Padoru.Core
 {
-    public class ScreenManager : IScreenManager
+    public class ScreenManager<TScreenId> : IScreenManager<TScreenId>
     {
-        private List<IScreen> screens = new List<IScreen>();
+        private readonly Dictionary<TScreenId, IScreen> screensDatabase = new();
+        
+        private IScreenProvider<TScreenId> provider;
+        private Canvas parentCanvas;
 
-        private IScreen currentScreen => screens.LastOrDefault();
-
-        public Canvas ParentCanvas { get; set; }
+        public void Init(IScreenProvider<TScreenId> providerReference, Canvas parentCanvasReference)
```

Comment:
Throw exceptions if the init params are null

Reply:
Fixed and add the test for the init method